packages/react-composites/src/composites/MeetingComposite/state/MeetingAdapterState.ts

```diff
@@ -71,3 +35,58 @@ export interface MeetingAdapterClientState extends Pick<CallAdapterClientState,
  * combined into one to suit the purpose of a Meeting.
  */
 export interface MeetingAdapterState extends MeetingAdapterUiState, MeetingAdapterClientState {}
+
+export function generateMeetingAdapterState(callAdapter: CallAdapter, chatAdapter: ChatAdapter): MeetingAdapterState {
+  const callAdapterState = callAdapter.getState();
+  const chatAdapterState = chatAdapter.getState();
+
+  const { call, displayName, userId, devices, page } = callAdapterState;
+  const meeting = call ? generateMeetingState(call, chatAdapterState.thread) : undefined;
+
+  return {
+    meeting,
+    userId,
+    page: callPageToMeetingPage(page),
+    displayName,
+    devices
+  };
+}
+
+export function mergeChatAdapterStateIntoMeetingAdapterState(
+  chatAdapterState: ChatState,
+  meetingAdapterState: MeetingAdapterState
+): MeetingAdapterState {
+  const newMeetingState = meetingAdapterState.meeting
+    ? mergeChatStateIntoMeetingState(chatAdapterState.thread, meetingAdapterState.meeting)
+    : undefined;
+
+  const { userId, page, displayName, devices } = meetingAdapterState;
```

Comment:
Simply `return {...meetingAdapterState, meeting: newMeetingState}`?

It's also future proof for adding fields to `MeetingAdatperState` that aren't affected by chat.

Reply:
Updated